rio/src/RIO/Prelude/Logger.hs

```diff
@@ -96,6 +106,13 @@ data LogFunc = LogFunc
   , lfOptions :: !(Maybe LogOptions)
   }
 
+-- | Pending documentation
+data ReloadableLogFunc = ReloadableLogFunc
+  { rlfLogFunc        :: !LogFunc
+  , rlfReloadLogFunc  :: !(LogOptions -> IO ())
```

Comment:
I would have thought you'd instead have something like:

```haskell
newtype ReloadableLogFunc = ReloadableLogFunc (IORef (LogFunc, IO ()))

updateLogFunc :: ReloadableLogFunc -> LogFunc -> IO () -> IO ()
updateLogFunc (ReloadableLogFunc ref) lf cleanup = mask_ $ do
  (_, cleanupOld) <- readIORef ref
  cleanupOld
  writeIORef ref (lf, cleanup)

rlfLogFunc :: ReloadableLogFunc -> LogFunc
rlfLogFunc (ReloadableLogFunc ref) a b c d = do
  (lf, _) <- readIORef ref
  lf a b c d
```

Reply:
I'm not opposed to this idea, but I believe there may be some changes needed to make this useful. The reason I used `LogOptions` as my input to reload, was because I can have that record outside of a `withLogFunc` block.

How would this update logic work if I only have a `withLogFunc`?

Reply:
Fairly poorly actually, but I think the whole approach has some problems around internal state. Let me try to reset the question a bit: is the log level the only thing we really need to be able to change? If so, I'd rather put just that in an `IORef` and make that directly modifiable via a setting to `LogOptions`.

Reply:
Ah good point, as far as I'm aware if I can change `logLevel` and `verbosity` I think this should be good.